frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts

```diff
@@ -217,12 +597,24 @@ export const ResourceYAMLEditor: React.FC<ResourceYAMLEditorProps> = require('@c
  * const [resource, loaded, loadError] = useK8sWatchResource(clusterResource);
  * return <ResourceEventStream resource={resource} />
  * ```
- * @param {ResourceEventStreamProps['resource']} - An object whose related events should be shown.
+ * @param {K8sResourceCommon} resource - An object whose related events should be shown.
  */
 export const ResourceEventStream: React.FC<ResourceEventStreamProps> = require('@console/internal/components/events')
   .WrappedResourceEventStream;
 
-const _usePrometheusPoll: (
+/**
+ * React hook to poll Prometheus for a single query.
+ * @param {string} [props.query] - (optional) Prometheus query string. If empty or undefined, polling is not started.
+ * @param {number} [props.delay] - (optional) polling delay interval (ms)
+ * @param {PrometheusEndpoint} props.endpoint - one of the PrometheusEndpoint (label, query, range, rules, targets)
+ * @param {number} [props.endTime] - (optional) for QUERY_RANGE enpoint, end of the query range
+ * @param {number} [props.samples] - (optional) for QUERY_RANGE enpoint
+ * @param {number} [options.timespan] - (optional) for QUERY_RANGE enpoint
+ * @param {string} [options.namespace] - (optional) a search param to append
+ * @param {string} [options.timeout] - (optional) a search param to append
+ * @returns A tuple containing the query response, a boolean flag indicating whether the response has completed, and any errors encountered during the request or post-processing of the request
+ */
+ const _usePrometheusPoll: (
```

Comment:
I think we should get rid of this one... if it's really needed, it either needs to drop the leading `_` or be moved to the `internal` package. This is not a good exposed API. See my [comment here](https://docs.google.com/document/d/1z0QCTxCr8hS8wDI5Oe3tmu7m3APNVMpmcTUDWjfDuS0/edit?disco=AAAAawSPw5A).

Ie, don't doc this. You may run into a conflict if you try to update something that is subsequently removed by another PR.